### PR TITLE
lnd: removing activeNetParams global, passed around instead via configs

### PR DESCRIPTION
**Pull request by Crypt-iQ**
This commit removes the `activeNetParams` global in `chainparams.go`. This is necessary to isolate code from the lnd package so we can import it for use in tests, other projects, etc.

### CODE REVIEW

**Comment by Crypt-iQ** [comment]
`TestFundingManagerMaxPendingChannels` is failing in `travis-race` but it is a known flake.

File: mock.go [hunk 1]
```diff
@@ -219,7 +219,7 @@ type mockChainIO struct {
 var _ lnwallet.BlockChainIO = (*mockChainIO)(nil)
 
 func (m *mockChainIO) GetBestBlock() (*chainhash.Hash, int32, error) {
-	return activeNetParams.GenesisHash, m.bestHeight, nil
+	return nil, m.bestHeight, nil
```
Comment (guggero) [inline mock.go:222]: Should we just return `testnetParams.GenesisHash` here to not change the behavior? Not sure it's even used now, though.

File: rpcserver.go [hunk 2]
```diff
@@ -502,6 +502,10 @@ type rpcServer struct {
 
 	// selfNode is our own pubkey.
 	selfNode route.Vertex
+
+	// rpcNetParams is the set of currently active parameters for the
+	// rpcServer.
+	rpcNetParams bitcoinNetParams
```
Comment (guggero) [inline rpcserver.go:508]: We have the full `*Config` available in `cfg`. Can just use that.

File: server.go [hunk 3]
```diff
@@ -274,6 +274,8 @@ type server struct {
 
 	hostAnn *netann.HostAnnouncer
 
+	serverNetParams bitcoinNetParams
```
Comment (guggero) [inline server.go:277]: We have `cfg` here which has the full config. This new field is not necessary.

File: server.go [hunk 4]
```diff
@@ -764,7 +768,7 @@ func newServer(cfg *Config, listenAddrs []net.Addr, chanDB *channeldb.DB,
 	s.authGossiper = discovery.New(discovery.Config{
 		Router:            s.chanRouter,
 		Notifier:          s.cc.chainNotifier,
-		ChainHash:         *activeNetParams.GenesisHash,
+		ChainHash:         *s.serverNetParams.GenesisHash,
```
Comment (guggero) [inline server.go:771]: Can just use `s.cfg.ActiveNetParams` here and below.

File: rpcserver.go [hunk 5]
```diff
@@ -987,7 +994,7 @@ func allowCORS(handler http.Handler, origins []string) http.Handler {
 func (r *rpcServer) sendCoinsOnChain(paymentMap map[string]int64,
 	feeRate chainfee.SatPerKWeight, label string) (*chainhash.Hash, error) {
 
-	outputs, err := addrPairsToOutputs(paymentMap)
+	outputs, err := addrPairsToOutputs(paymentMap, r.rpcNetParams.Params)
```
Comment (guggero) [inline rpcserver.go:997]: Can just use `r.cfg.ActiveNetParams` here and below.

File: mock.go (hunk 1 above)
Comment (Crypt-iQ) [reply mock.go:222]: I feel like this is fine because no tests failed in removing this

File: rpcserver.go (hunk 2 above)
Comment (Crypt-iQ) [reply rpcserver.go:508]: Yeah but eventually rpcserver will have its own config, guess it doesn't matter one way or the other at this stage.

File: server.go (hunk 3 above)
Comment (Crypt-iQ) [reply server.go:277]: See comment about rpcserver above.

File: chainparams.go [hunk 6]
```diff
@@ -10,10 +10,6 @@ import (
 	litecoinWire "github.com/ltcsuite/ltcd/wire"
 )
 
-// activeNetParams is a pointer to the parameters specific to the currently
-// active bitcoin network.
-var activeNetParams = bitcoinTestNetParams
```
Comment (carlaKC) [inline chainparams.go:15]: 🎉

File: config.go [hunk 7]
```diff
@@ -1329,13 +1335,13 @@ func extractBtcdRPCParams(btcdConfigPath string) (string, string, error) {
...
 
-// extractBitcoindParams attempts to extract the RPC credentials for an
+// extractBitcoindRPCParams attempts to extract the RPC credentials for an
 // existing bitcoind node instance. The passed path is expected to be the
 // location of bitcoind's bitcoin.conf on the target system. The routine looks
 // for a cookie first, optionally following the datadir configuration option in
 // the bitcoin.conf. If it doesn't find one, it looks for rpcuser/rpcpassword.
-func extractBitcoindRPCParams(bitcoindConfigPath string) (string, string, string,
-	string, error) {
+func extractBitcoindRPCParams(netParams bitcoinNetParams,
```
Comment (carlaKC) [inline config.go:1343]: nit: could just pass the `Params.Name` in here since it's all we use

File: chainparams.go [hunk 8]
```diff
@@ -10,10 +10,6 @@ import (
 	litecoinWire "github.com/ltcsuite/ltcd/wire"
 )
 
-// activeNetParams is a pointer to the parameters specific to the currently
```
Comment (halseth) [inline chainparams.go:13]: should we move this whole type out of the lnd package while we're at it?

Comment (Crypt-iQ) [reply chainparams.go:13]: Going to do this in a followup PR, but could also do it here

File: mock.go (hunk 1 above)
Comment (Crypt-iQ) [reply mock.go:222]: Could keep the same behavior but then we need to have `mockChainIO` store a `bitcoinNetParams` struct, or at least the genesis hash

File: rpcserver.go (hunk 2 above)
Comment (Crypt-iQ) [reply rpcserver.go:508]: Agree, will do the code move in a later PR

File: server.go (hunk 3 above)
Comment (Crypt-iQ) [reply server.go:277]: Agree

File: server.go (hunk 4 above)
Comment (Crypt-iQ) [reply server.go:771]: yup

File: rpcserver.go (hunk 5 above)
Comment (Crypt-iQ) [reply rpcserver.go:997]: yup

File: config.go (hunk 7 above)
Comment (Crypt-iQ) [reply config.go:1343]: Yeah, wasn't sure if we'd ever need it for anything else but we can cross that bridge if/when we get there

File: chainparams.go (hunk 8 above)
Comment (Crypt-iQ) [reply chainparams.go:13]: Going to do this in #4487

File: config.go [hunk 9]
```diff
@@ -1339,13 +1345,13 @@ func extractBtcdRPCParams(btcdConfigPath string) (string, string, error) {
...
 
-// extractBitcoindParams attempts to extract the RPC credentials for an
+// extractBitcoindRPCParams attempts to extract the RPC credentials for an
 // existing bitcoind node instance. The passed path is expected to be the
 // location of bitcoind's bitcoin.conf on the target system. The routine looks
 // for a cookie first, optionally following the datadir configuration option in
 // the bitcoin.conf. If it doesn't find one, it looks for rpcuser/rpcpassword.
-func extractBitcoindRPCParams(bitcoindConfigPath string) (string, string, string,
-	string, error) {
+func extractBitcoindRPCParams(paramName string,
```
Comment (guggero) [inline config.go:1353]: nit: call `networkName` instead of `paramName`?

File: mock.go (hunk 1 above)
Comment (guggero) [reply mock.go:222]: I don't think the default value of `activeNetParams` was overwritten in any of the tests that use the mock. So we really just could hard code the testnet value here to simplify things but keep the old behavior.

File: config.go (hunk 9 above)
Comment (Crypt-iQ) [reply config.go]: changed

File: mock.go (hunk 1 above)
Comment (Crypt-iQ) [reply mock.go]: Hardcoded testnet value

**Pull request merged**